src/aap_eda/tasks/ruleset.py

```diff
@@ -29,34 +30,98 @@
 
 
 @job("activation")
-def activate_rulesets(is_restart: bool, activation_id: int) -> None:
-    activation = models.Activation.objects.filter(id=activation_id).first()
-    if (
-        not activation
-        or str(activation.status) == ActivationStatus.DELETING.value
-    ):
-        logger.info(f"Activation id: {activation_id} is deleted")
-        return
+def activate(activation_id: int, requester: str = "User") -> None:
+    with transaction.atomic():
+        activation = (
+            models.Activation.objects.select_for_update()
+            .filter(id=activation_id)
+            .first()
+        )
+        if (
+            not activation
+            or str(activation.status) == ActivationStatus.DELETING.value
+        ):
+            logger.info(f"Activation id: {activation_id} is deleted")
+            return
 
-    # TODO(hsong): move this to views
-    if is_restart:
-        activation.restart_count += 1
-        activation.save(update_fields=["restart_count", "modified_at"])
-    logger.info(f"Task started: Activate rulesets ({activation.name})")
+        logger.info(
```

Comment:
Before this line we should validate activation's `is_enabled` is True and status is `pending`, otherwise log and return.